internal/rpcserver/interface.go

```diff
@@ -424,3 +424,43 @@ type Clock interface {
 	// Since returns the time elapsed since t.
 	Since(t time.Time) time.Duration
 }
+
+// FeeEstimator provides an interface that tracks historical data for published
+// and mined transactions in order to estimate fees to be used in new
+// transactions for confirmation within a target block window.
+//
+// The interface contract requires that all of these methods are safe for
+// concurrent access.
+type FeeEstimator interface {
+	// DumpBuckets returns the internal estimator state as a string.
+	DumpBuckets() string
```

Comment:
There is no need for all of these in the RPC interface.  There is only a single function that is needed, namely `EstimateSmartFee`.

The interface should have only exactly what is needed.  Since these are internal, we can change them if needed later without needing major version bumps.